core/src/main/java/org/jboss/logmanager/LogContext.java

```diff
@@ -43,7 +47,23 @@
  */
 @SuppressWarnings({"unused", "WeakerAccess"})
 public final class LogContext implements AutoCloseable {
-    private static final LogContext SYSTEM_CONTEXT = new LogContext(false);
+    private static final LogContext SYSTEM_CONTEXT = new LogContext(false, discoverDefaultInitializer());
+
+    private static LogContextInitializer discoverDefaultInitializer() {
+        final SecurityManager sm = System.getSecurityManager();
+        if (sm != null) {
+            return AccessController.doPrivileged((PrivilegedAction<LogContextInitializer>) LogContext::discoverDefaultInitializer0);
+        } else {
+            return discoverDefaultInitializer0();
+        }
+    }
+
+    private static LogContextInitializer discoverDefaultInitializer0() {
+        // allow exceptions to bubble up, otherwise logging won't work with no indication as to why
+        final ServiceLoader<LogContextInitializer> loader = ServiceLoader.load(LogContextInitializer.class, LogContext.class.getClassLoader());
+        final Iterator<LogContextInitializer> iterator = loader.iterator();
+        return iterator.hasNext() ? iterator.next() : LogContextInitializer.DEFAULT;
```

Comment:
Not really any update needed for this PR, but I wonder if it would make any sense to allow multiple services or have some kind of ranking. For example that would allow one in the`ext` project and if no others override it, it could use a `logging.properties` initializer of some sort.

Reply:
The way we have solved this case up until now has been to use class path ordering.  However this might not suffice when it comes to Quarkus, where (at present at least) there is a lot less control over the class path ordering.  So... yeah, this is something we might look into, for this as well as for configurators.

I think the `logging.properties` case should be solved by a configurator though.

Reply:
Cool. I wonder, maybe you guys don't use it, if we should look at having some kind of service loader like API with ranking order in wildfly-common.

Reply:
Looks like I've got something similar in the log manager already https://github.com/jboss-logging/jboss-logmanager/blob/f962451f68416608e055d895303e915dc530d493/core/src/main/java/org/jboss/logmanager/ConfiguratorFactory.java#L30

Reply:
I don't really anticipate that we'd have more than one initializer though.  It's not really the same thing as configuration where you might have multiple approaches; it's really tied to the run time.

I think it would be possible to add a priority value on the initializer interface later if we actually encountered a case where there could be more than one initializer; wdyt about that?

Reply:
Sounds good to me. I was mostly just brainstorming it.